Crypto/PubKey/ECIES.hs

```diff
@@ -33,16 +34,16 @@ import           Crypto.Internal.Proxy
 deriveEncrypt :: (MonadRandom randomly, EllipticCurveDH curve)
               => proxy curve -- ^ representation of the curve
               -> Point curve -- ^ the public key of the receiver
-              -> randomly (Point curve, SharedSecret)
+              -> randomly (CryptoFailable (Point curve, SharedSecret))
 deriveEncrypt proxy pub = do
     (KeyPair rPoint rScalar) <- curveGenerateKeyPair proxy
-    return (rPoint, ecdh proxy rScalar pub)
+    return $ (\s -> (rPoint, s)) `fmap` ecdh proxy rScalar pub
```

Comment:
it would probably make more sense to retry here instead of failing, since we control the scalar completely. it seems unlikely to happens either, so we could also ignore it, but retrying seems like the best thing to do here.